### PR TITLE
cmd(engine): optimize parameters of tiflow master and executor

### LINKED ISSUE #6965: Simplify command line parameters of tiflow binary
### Is your feature request related to a problem?

`tiflow master` and `tiflow executor` contains many parameters, part of them are difficult to configure from command line, such as metastore configuration. We should consider to move these parameters to config file only.

### Describe the feature you'd like

- [ ] Move part of command line parameters to config only
- [ ] Ensure all parameters can be configured by config file
- [ ] Make sure this update is compatible with all existing deployment tools, including helm in integration tests and tiflow operator

### Describe alternatives you've considered

_No response_

### Teachability, Documentation, Adoption, Migration Strategy

_No response_

### PR DESCRIPTION
**Pull request by CharlesCheung96**
### What problem does this PR solve?

Issue Number: close #6965 

### What is changed and how it works?
1. Remove ssl related params in cmd.
2. Ensure that parameter names are uniform.
3. Add `name` field to  `servermaster.Config` to facilitate cluster deployment.

#### Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
```bash
# ./tiflow master -h                                                                                           root@tiflow
Start a dataflow engine server master

Usage:
  tiflow master [flags]

Flags:
      --addr string                        Set the listening address for server master (default "127.0.0.1:10240")
      --advertise-addr string              Set the advertise listening address for client communication
      --business-meta-endpoints strings    business metastore endpoint (default [127.0.0.1:3336])
      --config string                      Path of the configuration file
      --etcd-endpoints strings             etcd endpoints (default [127.0.0.1:2379])
      --framework-meta-endpoints strings   framework metastore endpoint (default [127.0.0.1:3336])
  -h, --help                               help for master
      --log-file string                    log file path
      --log-level string                   log level (etc: debug|info|warn|error) (default "info")
      --name string                        human readable name for master

# ./tiflow executor -h                                                                                         root@tiflow
Start a dataflow engine executor

Usage:
  tiflow executor [flags]

Flags:
      --addr string             Set the listening address for executor (default "127.0.0.1:10340")
      --advertise-addr string   Set the advertise listening address for client communication
      --config string           Path of the configuration file
  -h, --help                    help for executor
      --join string             join to an existing cluster (usage: server masters' address) (default "127.0.0.1:10240")
      --labels stringToString   labels to the executor in key-value pairs (default [])
      --log-file string         log file path
      --log-level string        log level (etc: debug|info|warn|error) (default "info")
      --name string             human readable name for executor

```

### Release note 

```release-note
`None`.
```

### CODE REVIEW

File: deployments/engine/helm/tiflow/templates/cases.yaml [hunk 1]
```diff
@@ -28,6 +28,7 @@ spec:
             - "/tiflow-chaos-case"
             - "--duration=20m"
             - "--etcd-addr={{ include "tiflow.name" . }}-metastore-etcd:12479"
+            - "--business-meta-addr={{ include "tiflow.name" . }}-metastore-mysql:3306"
```
Comment (lonng) [inline deployments/engine/helm/tiflow/templates/cases.yaml:31]: Does this parameter duplicate with the configuration file?

File: engine/pkg/cmd/executor/executor.go [hunk 2]
```diff
@@ -55,21 +48,15 @@ func newOptions() *options {
...
 
-	cmd.Flags().StringVar(&o.executorConfig.LogConf.File, "log-file", o.executorConfig.LogConf.File, "log file path")
-	cmd.Flags().StringVar(&o.executorConfig.LogConf.Level, "log-level", o.executorConfig.LogConf.Level, "log level (etc: debug|info|warn|error)")
-
-	cmd.Flags().StringVar(&o.executorConfig.Name, "name", o.executorConfig.Name, "human readable name for executor")
 	cmd.Flags().StringVar(&o.executorConfig.Join, "join", o.executorConfig.Join, "join to an existing cluster (usage: server masters' address)")
 
 	cmd.Flags().StringVar(&o.executorConfigFilePath, "config", "", "Path of the configuration file")
-
-	cmd.Flags().StringVar(&o.caPath, "ca", "", "CA certificate path for TLS connection")
```
Comment (lonng) [inline engine/pkg/cmd/executor/executor.go:69]: TLS-related parameters are better to keep in the command line.

File: engine/pkg/cmd/master/master.go [hunk 3]
```diff
@@ -55,35 +48,17 @@ func newOptions() *options {
...
-	cmd.Flags().StringVar(&o.masterConfig.FrameMetaConf.Auth.User, "framework-meta-user", o.masterConfig.FrameMetaConf.Auth.User, "framework metastore user")
-	cmd.Flags().StringVar(&o.masterConfig.FrameMetaConf.Auth.Passwd, "framework-meta-password", o.masterConfig.FrameMetaConf.Auth.Passwd, "framework metastore password")
-	// NOTE: Schema Convention:
-	// 1.It needs to stay UNCHANGED for one dataflow engine cluster
-	// 2.It needs be different between any two dataflow engine clusters
-	// 3.Naming rule: https://dev.mysql.com/doc/refman/5.7/en/identifiers.html
-	cmd.Flags().StringVar(&o.masterConfig.FrameMetaConf.Schema, "framework-meta-schema", o.masterConfig.FrameMetaConf.Schema, `schema name for framework meta`)
-
 	cmd.Flags().StringSliceVar(&o.masterConfig.BusinessMetaConf.Endpoints, "business-meta-endpoints", o.masterConfig.BusinessMetaConf.Endpoints, "business metastore endpoint")
-	cmd.Flags().StringVar(&o.masterConfig.BusinessMetaConf.StoreType, "business-meta-store-type", o.masterConfig.BusinessMetaConf.StoreType, "business metastore store type")
```
Comment (lonng) [inline engine/pkg/cmd/master/master.go:75]: Good job.

**Comment by lonng** [comment]
This is a very good start for optimizing parameters. And I think our configuration file fields name should be optimized also.

File: deployments/engine/helm/tiflow/templates/cases.yaml (hunk 1 above)
Comment (CharlesCheung96) [reply deployments/engine/helm/tiflow/templates/cases.yaml:31]: Yes, but parameters in config file will be overwritten in case of conflict.

**Comment by CharlesCheung96** [comment]
> This is a very good start for optimizing parameters. And I think our configuration file fields name should be optimized also.

We will optimize this in the furture.

File: deployments/engine/helm/tiflow/values.yaml [hunk 4]
```diff
@@ -15,15 +15,19 @@ master:
   dataStorage: 2Gi
   logStorage: 1Gi
   config: |
-    [framework-metastore-conf]
-    auth.user = "root"
-    auth.passwd = ""
+    [framework-meta]
```
Comment (CharlesCheung96) [inline deployments/engine/helm/tiflow/values.yaml:18]: /cc @lichunzhu  Please make sure this update is compatible with tiflow operator.

File: engine/servermaster/config.go [hunk 5]
```diff
@@ -60,13 +64,14 @@ const (
...
 
+	Name          string `toml:"name" json:"name"`
 	Addr          string `toml:"addr" json:"addr"`
 	AdvertiseAddr string `toml:"advertise-addr" json:"advertise-addr"`
 
 	ETCDEndpoints []string `toml:"etcd-endpoints" json:"etcd-endpoints"`
 
-	FrameMetaConf    *metaModel.StoreConfig `toml:"framework-metastore-conf" json:"framework-metastore-conf"`
-	BusinessMetaConf *metaModel.StoreConfig `toml:"business-metastore-conf" json:"business-metastore-conf"`
+	FrameMetaConf    *metaModel.StoreConfig `toml:"framework-meta" json:"framework-meta"`
```
Comment (gozssky) [inline engine/servermaster/config.go:73]:
```suggestion
	FrameworkMeta    *metaModel.StoreConfig `toml:"framework-meta" json:"framework-meta"`
```

File: engine/servermaster/config.go [hunk 6]
```diff
@@ -60,13 +64,14 @@ const (
...
+	BusinessMetaConf *metaModel.StoreConfig `toml:"business-meta" json:"business-meta"`
```
Comment (gozssky) [inline engine/servermaster/config.go:74]:
```suggestion
	BusinessMeta *metaModel.StoreConfig `toml:"business-meta" json:"business-meta"`
```

File: deployments/engine/docker-compose/config/template_master.toml [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+
+
+[framework-meta]
+endpoints = ["127.0.0.1:3336"]  # Corresponding command line:  --framework-meta-endpoints
+user = "root"
+password = ""
+# NOTE: Schema Convention:
+#   1.It needs to stay UNCHANGED for one dataflow engine cluster
+# 	2.It needs be different between any two dataflow engine clusters
+# 	3.Naming rule: https://dev.mysql.com/doc/refman/5.7/en/identifiers.html
```
Comment (gozssky) [inline deployments/engine/docker-compose/config/template_master.toml:25]:
```suggestion
#   1.It needs to stay UNCHANGED for one dataflow engine cluster
#   2.It needs be different between any two dataflow engine clusters
#   3.Naming rule: https://dev.mysql.com/doc/refman/5.7/en/identifiers.html
```

**Review by gozssky — APPROVED** [review]

File: engine/servermaster/server.go [hunk 8]
```diff
@@ -609,30 +610,30 @@ func (s *Server) registerMetaStore(ctx context.Context) error {
...
 	}
-	s.businessClientConn, err = meta.NewClientConn(cfg.BusinessMetaConf)
+	s.businessClientConn, err = meta.NewClientConn(cfg.BusinessMeta)
 	if err != nil {
-		log.Error("connect to business metastore fail", zap.Any("config", cfg.BusinessMetaConf), zap.Error(err))
+		log.Error("connect to business metastore fail", zap.Any("config", cfg.BusinessMeta), zap.Error(err))
 		return err
 	}
-	log.Info("register business metastore successfully", zap.Any("metastore", cfg.BusinessMetaConf))
+	log.Info("register business metastore successfully", zap.Any("metastore", cfg.BusinessMeta))
```
Comment (amyangfei) [inline engine/servermaster/server.go:636]:
```suggestion
	log.Info("register business metastore successfully", zap.Any("config", cfg.BusinessMeta))
```

File: engine/servermaster/server.go [hunk 9]
```diff
@@ -609,30 +610,30 @@ func (s *Server) registerMetaStore(ctx context.Context) error {
 		log.Error("create framework meta client fail", zap.Error(err))
 		return err
 	}
-	log.Info("register framework metastore successfully", zap.Any("metastore", cfg.FrameMetaConf))
+	log.Info("register framework metastore successfully", zap.Any("metastore", cfg.FrameworkMeta))
```
Comment (amyangfei) [inline engine/servermaster/server.go:613]:
```suggestion
	log.Info("register framework metastore successfully", zap.Any("config", cfg.FrameworkMeta))
```

File: engine/servermaster/config.go [hunk 10]
```diff
@@ -39,18 +40,21 @@ const (
...
 	defaultMetricInterval    = 15 * time.Second
 	defaultMasterAddr        = "127.0.0.1:10240"
+	defaultEtcdEndpoint      = "127.0.0.1:2379"
 
 	// DefaultBusinessMetaID is the ID for default business metastore
 	DefaultBusinessMetaID        = "_default"
 	defaultBusinessMetaEndpoints = "127.0.0.1:3336"
 	defaultBusinessMetaUser      = "root"
 	defaultBusinessMetaPassword  = ""
+	defaultBusinessMetaSchema    = "test"
```
Comment (amyangfei) [inline engine/servermaster/config.go:50]: I find there exists same table in business meta and framework meta, it is better to use different default schema, such as `test_business` and `test_framework`

File: deployments/engine/docker/dev.Dockerfile [hunk 11]
```diff
@@ -1,9 +1,9 @@
 FROM alpine:3.16 as builder
 
-RUN apk add --no-cache curl dumb-init
+# RUN apk add --no-cache curl dumb-init
```
Comment (amyangfei) [inline deployments/engine/docker/dev.Dockerfile:3]: Please remove this line if it is useless.

Comment (CharlesCheung96) [inline deployments/engine/docker/dev.Dockerfile:3]:
```suggestion
RUN apk add --no-cache curl dumb-init
```

File: deployments/engine/docker/dev.Dockerfile [hunk 12]
```diff
@@ -1,9 +1,9 @@
...
 
 COPY ./bin/tiflow /
 COPY ./bin/tiflow-demoserver /tiflow-demoserver
 COPY ./bin/tiflow-chaos-case /tiflow-chaos-case
 
-ENTRYPOINT ["/usr/bin/dumb-init"]
+# ENTRYPOINT ["/usr/bin/dumb-init"]
```
Comment (CharlesCheung96) [inline deployments/engine/docker/dev.Dockerfile:9]:
```suggestion
ENTRYPOINT ["/usr/bin/dumb-init"]
```

File: engine/servermaster/config.go (hunk 10 above)
Comment (CharlesCheung96) [inline engine/servermaster/config.go:50]:
```suggestion
	defaultBusinessMetaSchema    = "test_business"
```

File: engine/servermaster/config.go [hunk 13]
```diff
@@ -39,18 +40,21 @@ const (
...
 
 	// FrameMetaID is the ID for frame metastore
 	FrameMetaID               = "_root"
 	defaultFrameMetaEndpoints = "127.0.0.1:3336"
 	defaultFrameMetaUser      = "root"
 	defaultFrameMetaPassword  = ""
+	defaultFrameMetaSchema    = "test"
```
Comment (CharlesCheung96) [inline engine/servermaster/config.go:57]:
```suggestion
	defaultFrameMetaSchema    = "test_framework"
```

**Review by amyangfei — APPROVED** [review]

File: deployments/engine/helm/tiflow/values.yaml (hunk 4 above)
Comment (lichunzhu) [reply deployments/engine/helm/tiflow/values.yaml:18]: okay, we will update this later

**Pull request merged**